bit_manipulation/quine_macluskey.cpp

```diff
@@ -0,0 +1,404 @@
+			for (unsigned int j = 0; j < epi[i].v.size(); j++) {		// the row numbers belonging to epi are separately stored in a vector.
+				select[epi[i].v[j]] = 0;
+			}
+		}
+		for (int i = 0; i < 16; i++) {
+			if (select[i] == 1) {
+				epirow.push_back(i);
+			}
+		}
+
+		bool areUEpi = false;
+		bool pushOk = true;
+		int duplicateN = 0;
+		BitNode additionEpi;
+		vector<vector<BitNode>> addEpi;	//additional epi
+		int count = 0;
+		for (unsigned int i = 0; i < pi.size(); i++) {
+			for (unsigned int j = 0; j < pi[i].v.size(); j++) {
+				for (unsigned int k = 0; k < epirow.size(); k++) {
+					if (epirow[k] == pi[i].v[j]) {
+						areUEpi = true;
+						additionEpi.v.push_back(pi[i].v[j]);
+					}
+				}
+			}
+			if (areUEpi) {
+				additionEpi.a = pi[i].a;
+				additionEpi.b = pi[i].b;
+				additionEpi.c = pi[i].c;
+				additionEpi.d = pi[i].d;
+				additionEpi.howmany1 = pi[i].howmany1;
+				sort(additionEpi.v.begin(), additionEpi.v.end());	// Sort to compare two vectors
+
+				/**
+				* @brief If the additional epi belongs to the existing epi, push_back is not performed.
+				* @details If 0 and 1 are included to be epi
+				* (0,2) cannot be pushed_back if (0,1) already exists in epi
+				*/
+				for (unsigned int l = 0; l < epi.size(); l++) {
+					if (epi[l].v.size() > additionEpi.v.size()) {
+						if (belongsToArr(epi[l].v, additionEpi.v, epi[l].v.size(), additionEpi.v.size())) {
+							pushOk = false;
+						}
+						else pushOk = true;
+					}
+					else if (epi[l].v.size() < additionEpi.v.size()) {
+						if (belongsToArr(additionEpi.v, epi[l].v, additionEpi.v.size(), epi[l].v.size())) {
+							epi.erase(epi.begin() + l);
+						}
+						pushOk = true;
+					}
+				}
+				if (pushOk) {
+					epi.push_back(additionEpi);
+				}
+			}
+			areUEpi = false;
+			pushOk = true;
+			additionEpi.v.clear();
+		}
+		return epi;
+	}
+
+	/**
+	* @brief function for print
+	* @param epi vector
+	* @return void
+	*/
+	void prettyprint(vector<BitNode>& epi) {
+		for (unsigned int i = 0; i < epi.size(); i++) {
+			if ((epi[i].a == 2) && (epi[i].b == 2) && (epi[i].c == 2) && (epi[i].d == 2)) {
+				printf("1");
+			}
+			if (epi[i].a == 1) {
+				printf("A");
+			}
+			else if (epi[i].a == 0) {
+				printf("A'");
+			}
+			if (epi[i].b == 1) {
+				printf("B");
+			}
+			else if (epi[i].b == 0) {
+				printf("B'");
+			}
+			if (epi[i].c == 1) {
+				printf("C");
+			}
+			else if (epi[i].c == 0) {
+				printf("C'");
+			}
+			if (epi[i].d == 1) {
+				printf("D");
+			}
+			else if (epi[i].d == 0) {
+				printf("D'");
+			}
+			if ((i < epi.size() - 1) && (!equalArrays(epi[i].v, epi[i + 1].v, min(epi[i].v.size(), epi[i + 1].v.size())))) {
+				printf(" + ");
+			}
+			else if ((i < epi.size() - 1) && (equalArrays(epi[i].v, epi[i + 1].v, min(epi[i].v.size(), epi[i + 1].v.size())))) {
+				printf(" or ");
+			}
+		}
+	}
+};
```

Comment:
Please add self-test cases so we can verify your program works properly (**see the [typical structure of a program](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#typical-structure-of-a-program) as a reference**). Let us know if you need any help here or in our [**Discord server**](https://discord.gg/c7MnfGFGa6). 😃